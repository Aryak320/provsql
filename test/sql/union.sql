\set ECHO none
\pset format unaligned
SET search_path TO provsql_test,provsql;

CREATE TABLE union_result AS
SELECT *,sr_formula(provenance(),'personnel_name') AS formula FROM (
  SELECT classification FROM personnel WHERE city='Paris'
  UNION
  SELECT classification FROM personnel
) t;

SELECT remove_provenance('union_result');
SELECT classification,replace(formula,'Paul ⊕ Nancy','Nancy ⊕ Paul') AS formula FROM union_result ORDER BY classification;
DROP TABLE union_result;

CREATE TABLE union_result AS
SELECT *,formula(provenance(),'personnel_name') AS formula FROM (
  SELECT FROM personnel WHERE city='Paris'
  UNION
  SELECT FROM personnel
) t;

SELECT remove_provenance('union_result');
SELECT * FROM union_result;
DROP TABLE union_result;
