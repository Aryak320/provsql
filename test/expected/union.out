\set ECHO none
remove_provenance

classification,formula
unclassified,John
restricted,(Nancy ⊕ Nancy ⊕ Paul)
confidential,(Dave ⊕ Dave)
secret,(Ellen ⊕ Susan)
top_secret,(Magdalen ⊕ Magdalen)
